shared/macros-oval.jinja

```diff
@@ -515,3 +515,72 @@
 
   </def-group>
 {{%- endmacro %}}
+
+{{#
+  Macro which generates OVAL definition, test and object that check if 
+  a value of certain type is present in a YAML (or JSON) file at a given path.
+  Parameters:
+    - path - path of the file to check
+    - file - name of the file to check
+    - yamlpath - OVAL's YAML Path expression
+    - value - the value to check
+    - type (string/SimpleDatatypeEnumeration) - datatype for state's value_of, default is 'string'
+    - check (string/CheckEnumeration) - entity_check value for state's value_of, default is 'all'
+    - negate (boolean) - invert the meaning of the value check criterion, default false
+    - ocp_data (boolean) - the file is the part of the dump of OCP configuration, take into account the 'ocp_data_root' variable
+#}}
+{{%- macro oval_yamlfile_value(path='', file='', yamlpath='', value='', type='', check='', negate=false, ocp_data=false) -%}}
+{{% if target_oval_version >= [5, 11] %}}
+  <def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="1">
+      <metadata>
+        <title>Check the value of '{{{ yamlpath }}}' in the "{{{ file }}}" file</title>
+        {{{- oval_affected(products) }}}
+        <description>The file {{{ file }}} in {{{ path }}} should {{% if negate %}}not{{% endif %}} contain value "{{{ value }}}" of type {{{ type }}} at {{{ yamlpath }}}.</description>
+      </metadata>
+      <criteria>
+          <criterion comment="Make sure that {{% if negate %}}no{{% endif %}} {{% if check %}}({{{ check }}}){{% endif %}} element of '{{{ yamlpath }}}' has value '{{{ value }}}'' of type {{{ type }}}." {{% if negate %}}negate="true"{{% endif %}} test_ref="test_{{{ rule_id }}}"/>
+          <criterion comment="Make sure that there is the file '{{{ file }}}' exists in {{{ path }}}." test_ref="test_file_for_{{{ rule_id }}}"/>
+      </criteria>
+    </definition>
+
+    <local_variable id="{{{ rule_id }}}_file_location" datatype="string" comment="The actual path of the file to scan." version="1">
+       {{% if ocp_data %}}
+       <concat>
+	       <variable_component var_ref="ocp_data_root"/>
+	       <literal_component>{{{ path }}}</literal_component>
+       </concat>
+       {{% else %}}
+       <literal_component>{{{ path }}}</literal_component>
+       {{% endif %}}
+    </local_variable>
+
+    <ind:yamlfilecontent_test id="test_{{{ rule_id }}}" check="at least one" comment="Find at least one match of '{{{ value }}}' in '{{{ yamlpath }}}'." version="1">
```

Comment:
> `check="at least one"`

If multiple objects are found is it enough that at least one conforms to the state? Can the others be non-compliant?

Reply:
I'm not sure if there is a way to fetch multiple objects for this probe. I'll check the code.

Reply:
So I think that for this template it would make sense to have 'only one' there.

Reply:
Looks good.